### DISCUSSION TITLE
Duplicate Row - Action Function Issues

### DISCUSSION THREAD

**Discussion by Mcc83** [Budibase/budibase#12747] [Help]
Having a little bit of a headscratcher here.

I am attempting to have a Delete button that just copies the rows data into another database (List of Deleted Items) and then deletes the row of the first database it was in.

This is so that I can delete a row (they are inventory items for my case) and then I'll have a navigation link to where I can view all of my deleted items. That way I can view those items and either undelete them via the same process in reverse or Permanently delete them.

I tried using the duplicate row action as a new button in a form block and can not successfully write to the other database. Does the backup sister database need to be an exact copy of columns before I can write to it? A new fresh database or a duplicate without any rows does not change my results. 

If someone can provide an example model of this simple back and forth of rows between two identical databases (in regards to columns not rows) I could probably figure out what I'm doing wrong.

Currently not receiving any errors, I am only aware that the rows are not being written in the new database.

**Comment by melohagan** [comment]
Hey @Mcc83 

If you want to copy a row from one table to another, use the **Save Row** action, not the duplicate action. The duplicate row action is only for duplicating a row within the same table.

![Screenshot 2024-01-10 at 09 49 59](https://github.com/Budibase/budibase/assets/101575380/c076663e-5a53-454b-a235-c3f7c1b1f061)

So in that screenshot you can see the original table has a details side panel with a 'clone' button that simply saves the row to the clone table.
*Datasource* is set to 'None' as we want a brand new `_id` to be generated, but that does also mean we need to use the `Add Column` button to specify the table columns and bind them to the fields of the original table row we are copying, e.g. `{{ Details form block.Employees.First Name }}`

https://github.com/Budibase/budibase/assets/101575380/e1a7a024-d9fd-4581-baf1-d6c1e78fde9d

App export: 
[Clone row-export.tar.gz](https://github.com/Budibase/budibase/files/13885925/Clone.row-export.tar.gz)

**Comment by Mcc83** [comment]
Thanks @melohagan , Very helpful! Sorry for taking a month+ to get back to you my work has me jumping around on different projects and I've just had time to get back to this.

I'm still having issues writing to my new database. My action button seems to create a new entry in the desired table (DeletedInventory) but does not write any data from my original table (ActiveInventory) to it. Even after binding the fields. Would you mind taking a quick a look if you have the time?

I'm going to see if I can figure this out before you respond and I'll edit out the rest of my comment here.

Attached is a copy of my app. You'll find the function we're working on under the Search page and buried in there is a buttongroup. Any insight would be much appreciated. 

Password for app is :password
Never exported prior to this so wasn't sure how to get around password requirement.
[Nacol2-export-1708461037460.enc.tar.gz](https://github.com/Budibase/budibase/files/14350891/Nacol2-export-1708461037460.enc.tar.gz)
